main.tf

```
@@ -0,0 +1,107 @@
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = ">= 3.8"
+    }
+  }
+  backend "azurerm" {
+    resource_group_name  = "KPMG21_AbdulraoufAtia_ProjectExercise"
+    storage_account_name = "m12storageaccount"
+    container_name       = "m12containertf"
+    key                  = "terraform.tfstate"
+  }
+}
+
+provider "azurerm" {
+  features {}
+}
+
+data "azurerm_resource_group" "main" {
+  name = "KPMG21_AbdulraoufAtia_ProjectExercise"
+}
+
+resource "azurerm_service_plan" "main" {
+  name                = "${var.prefix}-terraformed-asp"
+  location            = data.azurerm_resource_group.main.location
+  resource_group_name = data.azurerm_resource_group.main.name
+  os_type             = "Linux"
+  sku_name            = "B1"
+}
+
+resource "azurerm_linux_web_app" "main" {
+  name                = "ARA-TODO-APP"
+  location            = data.azurerm_resource_group.main.location
+  resource_group_name = data.azurerm_resource_group.main.name
+  service_plan_id     = azurerm_service_plan.main.id
+  site_config {
+    application_stack {
+      docker_image     = "abdulraoufatia/todoapp"
+      docker_image_tag = "latest"
+    }
+  }
+
+  app_settings = {
+    "DOCKER_REGISTRY_SERVER_URL"          = "https://index.docker.io",
+    "CLIENTID"                            = "${var.CLIENTID}",
+    "CLIENTSECRET"                        = "${var.CLIENTSECRET}",
+    "FLASK_APP"                           = "todo_app/app:create_app",
+    "ID"                                  = "${var.ID}",
+    "PRIMARY_CONNECTION_STRING"           = "${var.PRIMARY_CONNECTION_STRING}",
```

We shouldn't be using a variable for the connection string - we want Terraform to tell _us_ what that is, since we don't know before the DB has been created!
I'd expect this to be more like below so that Terraform can read in the appropriate value:
```suggestion
    "PRIMARY_CONNECTION_STRING"           = azurerm_cosmosdb_account.db.connection_strings[0],
```